pkg/planner/core/task.go

```diff
@@ -544,6 +542,31 @@ func (p *PhysicalHashJoin) attach2TaskForMpp(tasks ...task) task {
 		partTp:   outerTask.partTp,
 		hashCols: outerTask.hashCols,
 	}
+	defaultSchema := BuildPhysicalJoinSchema(p.JoinType, p)
+	if p.schema.Len() < defaultSchema.Len() {
+		if p.schema.Len() > 0 {
```

Comment:
We'd better column pruning in logical rule  phase rather than attachToTask .

Reply:
I think, since this work takes effect only for mpp task, it seems reasonable to add it here.